### ISSUE TITLE
Bug: Normalizer breaks mapper

### ISSUE THREAD

**Issue by mastir** [CuyZ/Valinor#572]
hey @romm, today i spent 6 hours to find rare random crush source in my project. Long story short: if `$normalizer->normalize()` used before `$mapper->map` on class with enum property, mapper unable to map objects of this class. Here is minimal test case:

```php
use CuyZ\Valinor\MapperBuilder;
use CuyZ\Valinor\Normalizer\Format;
use PHPUnit\Framework\TestCase;

enum AEnum : int
{
    case ONE = 1;
    case TWO = 2;
}

class AClass {

    public function __construct(public readonly AEnum $value)
    {
    }
}

class BugTest extends TestCase
{

    public function setUp(): void
    {
        $builder = new MapperBuilder();
        $this->mapper = $builder->mapper();
        $this->normalizer = $builder->normalizer(Format::array());
    }

    public function testWorks()
    {
        $this->mapper->map(AClass::class, [ 'value' => 1 ]);
        $this->assertTrue(true);
    }

    public function testFails()
    {
        $this->normalizer->normalize(new AClass(AEnum::ONE));
        $this->testWorks();
    }


}
```

testWorks - works
testFails - ```CuyZ\Valinor\Mapper\TypeTreeMapperError: Could not map type `Module\ModelMaker\tests\AClass`. An error occurred at path value: Value 1 does not match type `array{name: string, value: int}`.```

Any suggestions on deeper investigation?

**Comment by romm** [comment]
Hey @mastir I'm very sorry about that, this issue has already been fixed with #565 but was not released yet.

I'll take some time tomorrow to publish a new release.

Thanks for taking time to investigate.

**Comment by mastir** [comment]
Nice. Thank you!
